trinity/extensibility/plugin.py

```diff
@@ -70,6 +156,35 @@ def stop(self) -> None:
         pass
 
 
+class BaseIsolatedPlugin(BasePlugin):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._process: Process = None
```

Comment:
This seems like it could be a class level attribute allowing you to remove this `__init__` method entirely.

Reply:
wait, but then it becomes bound to that class and not the instance, no? We will have many instances of such plugins and each needs to address its very own process instance. Am I understanding you wrong?

Reply:
Oh wow, I just realized that my mental model for how class attributes work was still wrong. I just read through this [super detailed guide](https://www.toptal.com/python/python-class-attributes-an-overly-thorough-guide) and now I understand.

So, while it is fine to have `_process: Process = None` as a class attribute **and still maintain different `_process` instances on the instance level** it would fall apart to have something like `_process_list: List[Process] = []` on the class level if the reference isn't *overwritten* but *mutated* in instances e.g. by  `self._process_list.append(some_process)`. In that case, that process would appear in the `_process_list` of all other plugin instances as well.

With that in mind, class level attributes have more legitimate usage that I thought they have.

Reply:
I'm just posting this here for anyone who (as me) did not know this. I'm aware this won't be news for most of you reading this but maybe it will help someone else.

```python
from typing import (
    List
)
from multiprocessing import (
    Process
)

class Foo:

    # FINE
    _proc: Process = None

    # BAD! DON'T DO THIS
    _proc_list: List[Process] = []
    
    def do_something(self):
        self._proc = Process()
        self._proc_list.append(self._proc)


foo1 = Foo()
foo1.do_something()

foo2 = Foo()
foo2.do_something()


print("shared _proc instance? {}".format(foo1._proc is foo2._proc))
print("shared _proc_list instance? {}".format(foo1._proc_list is foo2._proc_list))
```

prints:

```
$ python class_attributes.py 
shared _proc instance? False
shared _proc_list instance? True
```